### PR TITLE
issue 766

### PR DESCRIPTION
**Pull request by nmccready**
#766

### CODE REVIEW

**Comment by nmccready** [comment]
@jessertaylor @tombatossals @zacronos it would be nice if the spec passed and was not using `xit` , but I'm tired of looking at it. I know the example works.

**Comment by nmccready** [comment]
If anyone wants to make that spec pass please do.

File: src/directives/markers.js [hunk 1]
```diff
@@ -114,10 +116,12 @@ angular.module("leaflet-directive").directive('markers',
             }
         }
     };
-    var _destroy = function(markerModels, oldMarkerModels, lMarkers, map, layers){
-        // Delete markers from the array
+    var _seeWhatWeAlreadyHave = function(markerModels, oldMarkerModels, lMarkers, isEqual, cb){
```
Comment (nmccready) [inline src/directives/markers.js]: This function is mostly what _destroy did and is the functionality that I said needed to be in _addMarkers (called before). The difference being that on adding it looks for matches and on _destroy it looks for differences.

**Comment by tombatossals** [comment]
Great job @nmccready, I'll take a look to the test.

**Comment by nmccready** [comment]
@tombatossals btw I was looking at the many markers example now vs older releases. It looks like the time it takes to render the 10,000 markers increased ~2.5-3 seconds. I think this is from the generic event handling logic, but I am not entirely sure.

One thing to note is `for in` is being used everywhere as well as by our `$it.each`.  see http://jsperf.com/iterators/3

**Pull request merged**